monitor.cpp

```diff
@@ -3,36 +3,53 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
+#include <string>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+constexpr float MAX_TEMP = 102.0;
+constexpr float MIN_TEMP = 95.0;
+constexpr int MAX_PULSE = 100;
+constexpr int MIN_PULSE = 60;
+constexpr float MIN_SPO2 = 90.0;
+
+
+void displayAlert(const std::string& message) {
+    cout << message << "\n";
+    std::string blinkPattern[2] = {"\r* ", "\r *"};
+    for (int i = 0; i < 6; ++i) {
+        cout << blinkPattern[i % 2] << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+    cout << "\n";
+}
+
+bool isTemperatureCritical(const Vitals& vitals) {
+    return vitals.temperature > MAX_TEMP || vitals.temperature < MIN_TEMP;
+}
+
+bool isPulseRateCritical(const Vitals& vitals) {
+    return vitals.pulseRate < MIN_PULSE || vitals.pulseRate > MAX_PULSE;
+}
+
+bool isSpo2Critical(const Vitals& vitals) {
+    return vitals.spo2 < MIN_SPO2;
+}
+
+int vitalsOk(const Vitals& vitals) {
+    VitalCheck checks[] = {
+        {isTemperatureCritical, "Temperature is critical!"},
+        {isPulseRateCritical, "Pulse Rate is out of range!"},
+        {isSpo2Critical, "Oxygen Saturation is out of range!"}
+    };
+
+    bool allVitalsOk = true;
+
+    for (const auto& check : checks) {
```

Comment:
Nice way to summarize the result. Adding more vitals does not change the summarization code :)